File: 2831-number-of-beautiful-pairs/2831-number-of-beautiful-pairs.py
class Solution:
    def countBeautifulPairs(self, nums: List[int]) -> int:
        count=0
        for i in range (len(nums)):
            num1=str(nums[i])[0]
            for j in range (i+1,len(nums)):
                num=str(nums[j])
                nums2=num[len(num)-1]
                if(gcd(int(num1),int(nums2))==1):
                    count+=1
        return count

File: 2831-number-of-beautiful-pairs/2831-number-of-beautiful-pairs.cpp
class Solution {
public:
    int countBeautifulPairs(vector<int>& nums) {
        int count=0;
        for(int i=0;i<nums.size();i++){
            for(int j=i+1;j<nums.size();j++){
                string nums1=to_string(nums[i]);
                string nums2=to_string(nums[j]);
                int num1=nums1[0]-'0';
                int num2=nums2.back()-'0';
                if(gcd(num1,num2)==1){
                    count++;
                }
            }
        }
        return count;
    }
};